### PR TITLE
New deploy bugs

### PR DESCRIPTION
**Pull request by AtomicManic**
I've made changes across many files:
1. Removed unnecessary code 
2. added tested true/false to the variant returned
3. removed bson
4. removed inc call count where it wasn't needed

### CODE REVIEW

File: Service/utils.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 const geoip = require("geoip-lite");
 const parser = require("ua-parser-js");
-const requestIp = require("request-ip");
```
Comment (sufka861) [inline Service/utils.js:3]: are you sure we dont need to keep this?

File: controller/external.controller.js [hunk 2]
```diff
@@ -69,18 +69,19 @@ const experimentExistingUser = async (
...
 };
 
-const doExperiment = async (experimentId, uuid, req) => {
+const doExperiment = async (experimentId, uuid) => {
   await ExperimentRepository.incCallCount(experimentId);
-  let variant = await checkExperimentTypeAndExecExperiment(experimentId, req);
+  let variant = await checkExperimentTypeAndExecExperiment(experimentId);
   const userExperiment = { experimentId, variant };
   const updatedUser = await insertExperiment(uuid, userExperiment);
+  if(!updatedUser) throw new ServerUnableError('insert experiment to user')
```
Comment (sufka861) [inline controller/external.controller.js:84]: good validation

File: controller/user.controller.js [hunk 3]
```diff
@@ -49,11 +46,6 @@ const getUserExperiment = (user) => {
   return user.experiments;
 };
 
-const setCookie = () => {
```
Comment (sufka861) [inline controller/user.controller.js:52]: why are you deleting this?

File: Service/utils.js (hunk 1 above)
Comment (AtomicManic) [reply Service/utils.js:3]: yes, that was the thing that need the whole request

File: controller/user.controller.js [hunk 4]
```diff
@@ -49,11 +46,6 @@ const getUserExperiment = (user) => {
...
-  const uuid = generateUuid();
```
Comment (sufka861) [inline controller/user.controller.js:53]: same, why delete?

File: controller/user.controller.js (hunk 3 above)
Comment (AtomicManic) [reply controller/user.controller.js:52]: I am not using this. I made this function and than used a different logic as the main flow progressed

File: controller/user.controller.js (hunk 4 above)
Comment (AtomicManic) [reply controller/user.controller.js:53]: we don't need this for the same reason as the last one.

**Review by sufka861 — APPROVED** [review]
looks good
didnt undersrand though why delete
- generateUUID
- setCookie
- requestIP
if you ment to delete those than ok

**Pull request merged**